### PR TITLE
2.3.1 re-merge

### PR DESCRIPTION
**Pull request by onetechnical**
## Summary

Re-merge of master post-2.3.1 commits.

## Test Plan

Standard testing, but mostly verify changes to develop.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ addons:
     - python3-setuptools
     - python3-wheel
 script:
+- echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
```
Comment (onetechnical) [inline .travis.yml:20]:
```suggestion
```

Comment (onetechnical) [reply .travis.yml:20]: This is handled by moving it higher up in the file.

File: Makefile [hunk 2]
```diff
@@ -21,7 +21,7 @@ idb/postgres/setup_postgres_sql.go:	idb/postgres/setup_postgres.sql
 types/protocols_json.go:	types/protocols.json types/consensus.go
 	cd types && go generate
 
-idb/mocks/IndexerDb.go:	idb/dummy.go
+mocks:
```
Comment (onetechnical) [inline Makefile:24]:
```suggestion
idb/mocks/IndexerDb.go:	idb/dummy.go
```

File: go.mod [hunk 3]
```diff
@@ -17,5 +17,6 @@ require (
 	github.com/spf13/viper v1.7.1
 	github.com/stretchr/testify v1.6.1
 	github.com/valyala/fasttemplate v1.2.0 // indirect
+	github.com/vektra/mockery v1.1.2 // indirect
```
Comment (onetechnical) [inline go.mod:20]:
```suggestion
```

File: go.sum [hunk 4]
```diff
@@ -449,6 +449,8 @@ github.com/valyala/fasttemplate v1.1.0 h1:RZqt0yGBsps8NGvLSGW804QQqCUYYLsaOjTVHy
 github.com/valyala/fasttemplate v1.1.0/go.mod h1:UQGH1tvbgY+Nz5t2n7tXsz52dQxojPUpymEIMZ47gx8=
 github.com/valyala/fasttemplate v1.2.0 h1:y3yXRCoDvC2HTtIHvL2cc7Zd+bqA+zqDO6oQzsJO07E=
 github.com/valyala/fasttemplate v1.2.0/go.mod h1:UQGH1tvbgY+Nz5t2n7tXsz52dQxojPUpymEIMZ47gx8=
+github.com/vektra/mockery v1.1.2 h1:uc0Yn67rJpjt8U/mAZimdCKn9AeA97BOkjpmtBSlfP4=
```
Comment (onetechnical) [inline go.sum:452]:
```suggestion
```

**Review by winder — CHANGES_REQUESTED** [review]
Aside from the version file, this change should be a no-op

File: go.sum [hunk 5]
```diff
@@ -449,6 +449,8 @@ github.com/valyala/fasttemplate v1.1.0 h1:RZqt0yGBsps8NGvLSGW804QQqCUYYLsaOjTVHy
...
+github.com/vektra/mockery v1.1.2/go.mod h1:VcfZjKaFOPO+MpN4ZvwPjs4c48lkq1o3Ym8yHZJu0jU=
```
Comment (onetechnical) [inline go.sum:453]:
```suggestion
```

File: go.sum [hunk 6]
```diff
@@ -681,6 +683,7 @@ golang.org/x/tools v0.0.0-20200224181240-023911ca70b2/go.mod h1:TB2adYChydJhpapK
 golang.org/x/tools v0.0.0-20200227222343-706bc42d1f0d/go.mod h1:TB2adYChydJhpapKDTa4BR/hXlZSLoq2Wpct/0txZ28=
 golang.org/x/tools v0.0.0-20200304193943-95d2e580d8eb/go.mod h1:o4KQGtdN14AW+yjsvvwRTJJuXz8XRtIHtEnmAXLyFUw=
 golang.org/x/tools v0.0.0-20200312045724-11d5b4c81c7d/go.mod h1:o4KQGtdN14AW+yjsvvwRTJJuXz8XRtIHtEnmAXLyFUw=
+golang.org/x/tools v0.0.0-20200323144430-8dcfad9e016e/go.mod h1:Sl4aGygMT6LrqrWclx+PTx3U+LnKx/seiNR+3G19Ar8=
```
Comment (onetechnical) [inline go.sum:686]:
```suggestion
```

**Review by onetechnical — COMMENTED** [review]
Other variations picked up; develop should be primary.

**Review by winder — APPROVED** [review]
LGTM

**Pull request merged**